client/src/client.rs

```diff
@@ -52,39 +52,45 @@ pub struct Client {
 /// Representation of `Iroha` client.
 impl Client {
     /// Constructor for client from configuration
-    pub fn new(configuration: &Configuration) -> Self {
+    ///
+    /// # Errors
+    /// If configuration isn't valid (e.g public/private keys don't match)
+    pub fn new(configuration: &Configuration) -> Result<Self> {
         Self::with_headers(configuration, http_client::Headers::default())
     }
 
     /// Constructor for client from configuration and headers
     ///
     /// *Authentication* header will be added, if `login` and `password` fields are presented
+    ///
+    /// # Errors
+    /// If configuration isn't valid (e.g public/private keys don't match)
     pub fn with_headers(
         configuration: &Configuration,
         mut headers: HashMap<String, String>,
-    ) -> Self {
+    ) -> Result<Self> {
         if let Some(basic_auth) = &configuration.basic_auth {
             let credentials = format!("{}:{}", basic_auth.web_login, basic_auth.password);
             let encoded = base64::encode(credentials);
             headers.insert(String::from("Authorization"), format!("Basic {}", encoded));
         }
 
-        Self {
+        Ok(Self {
             torii_url: configuration.torii_api_url.clone(),
             telemetry_url: configuration.torii_telemetry_url.clone(),
             transaction_limits: configuration.transaction_limits,
             key_pair: KeyPair::new(
                 configuration.public_key.clone(),
                 configuration.private_key.clone(),
-            ),
+            )?,
```

Comment:
Maybe just store `KeyPair` in `configuration`? So that `Client` doesn't need to return `Result`?

Reply:
I think i's not stored as a key pair because that would affect environment variables and deserialization of configuration. I'd be happy to open an issue about this

Reply:
I opened an issue #2135 for it

Reply:
We should keep in mind that providing a private key automatically implies the keypair. The public key is an optional sanity check to keep track of which key was used in which place.